samples/mimxrt1060/basic-sample/src/iotconnect_app.c

```diff
@@ -173,29 +219,107 @@ bool app_startup(NX_IP *ip_ptr, NX_PACKET_POOL *pool_ptr, NX_DNS *dns_ptr) {
+        printf("Obtaining the operational certificate via DDIM.\r\n");
+        if (iotcdi_obtain_operational_identity(&azrtos_config, &ddim_interface, auth_context, config->env)) {
+            printf("Failed to obtain the operational certificate via DDIM.\r\n");
+        }
+    //}
+    if (false == extract_cpid_and_duid_from_operational_cn(config, ddim_interface.extract_operational_cn(auth_context))) {
+    	return false;
+    }
+
 #else
     extern const UCHAR sample_device_cert_ptr[];
     extern const UINT sample_device_cert_len;
     extern const UCHAR sample_device_private_key_ptr[];
     extern const UINT sample_device_private_key_len;
-    config->auth.type = IOTC_X509_RSA;
-    config->auth.data.identity.client_private_key = (UCHAR*) sample_device_private_key_ptr;
-    config->auth.data.identity.client_private_key_len = sample_device_private_key_len;
-    config->auth.data.identity.client_certificate = (UCHAR*) sample_device_cert_ptr;
-    config->auth.data.identity.client_certificate_len = sample_device_cert_len;
-#endif
+	struct sw_der_driver_parameters dp = {0};
+	dp.cert = (uint8_t *) (sample_device_cert_ptr);
+	dp.cert_size = sample_device_cert_len;
+	dp.key = (uint8_t *) (sample_device_private_key_ptr);
+	dp.key_size = sample_device_private_key_len;
+	dp.crypto_method = &crypto_method_ec_secp384;
+	if(create_sw_der_auth_driver( //
+	    		&(config->auth.data.x509.auth_interface), //
+				&(config->auth.data.x509.auth_interface_context), //
+				&dp)) { //
+	    	return false;
+	    }
+	auth_driver_context = config->auth.data.x509.auth_interface_context;
 
-    azrtos_config.ip_ptr = ip_ptr;
-    azrtos_config.pool_ptr = pool_ptr;
-    azrtos_config.dns_ptr = dns_ptr;
+#endif // ENABLE_DDIM_TO_DRIVER_SAMPLE
+    config->auth.type = IOTC_X509;
+#endif // IOTCONNECT_SYMETRIC_KEY
 
     while (true) {
+#ifdef MEMORY_TEST
+        // check for leaks
+        memory_test();
+#endif //MEMORY_TEST
         if (iotconnect_sdk_init(&azrtos_config)) {
+#ifdef ENABLE_DDIM_TO_DRIVER_SAMPLE
+        	// We should try obtain a new operational identity
+        	IotclSyncResult sync_result = iotconnect_get_last_sync_result();
+        	if (IOTCL_SR_DEVICE_NOT_REGISTERED == sync_result || IOTCL_SR_DEVICE_INACTIVE == sync_result) {
+                if (iotcdi_obtain_operational_identity(&azrtos_config, &ddim_interface, auth_context, config->env)) {
+                	printf("Operational certificate is missing. Obtaining it via DDIM.\r\n");
+                	return false;
+                }
+				if (false == extract_cpid_and_duid_from_operational_cn(config, ddim_interface.extract_operational_cn(auth_context))) {
+					return false;
+				}
+				printf("Obtained new operational certificate from DDIM.\r\n");
+				continue;
+        	}
+#endif
             printf("Unable to establish the IoTConnect connection.\r\n");
             return false;
         }
         // send telemetry approximately ever 5 seconds for 5 minutes
```

Comment:
Is comment still correct?

Reply:
Applied a few consistency fixes across iotconnect_app.c. We tend to change those delay s often when testing stuff so impossible to keep that comment up to date, so made it generic.